core/app/models/spree/order_inventory.rb

```diff
@@ -74,6 +74,8 @@ def add_to_shipment(shipment, quantity)
         shipment.stock_location.unstock(variant, quantity, shipment)
       end
 
+      line_item.reload
```

Comment:
I'm not sure if this is necessary, but adding reloads generally wreaks havoc. `reload` disconnects the object graph between models, which can cause some very unusual behaviour. Instead we should look at fixing whatever code was updating this line item (and not uploading the correct version, which we have here). A spec would be helpful since removing this line causes no failures